executor/executor.go

```diff
@@ -329,14 +329,34 @@ func (e *ShowDDLExec) Next(ctx context.Context, req *chunk.Chunk) error {
 // ShowDDLJobsExec represent a show DDL jobs executor.
 type ShowDDLJobsExec struct {
 	baseExecutor
+	DDLJobExecInitializer
 
-	cursor         int
+	cacheJobs []*model.Job
+	jobNumber int
+	is        infoschema.InfoSchema
+	done      bool
+}
+
+// DDLJobExecInitializer initials the DDLJobsExec.
+type DDLJobExecInitializer struct {
```

Comment:
No need to export `DDLJobExecInitializer`?

Reply:
BTW, How about rename to `DDLJobRetriever`?

Reply:
done